src/test/java/org/jenkinsci/plugins/workflow/cps/PersistenceProblemsTest.java

```diff
@@ -337,4 +338,25 @@ public void inProgressButStartBlocksLost() throws Exception {
             assertCompletedCleanly(run);
         });
     }
+
+    @Issue("JENKINS-50888")  // Tried to modify build without lazy load being triggered
+    @Test public void modifyBeforeLazyLoad() {
+        story.thenWithHardShutdown(r -> {  // Normal build
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition("echo 'dosomething'", true));
+            r.buildAndAssertSuccess(p);
+        });
+        story.then(r -> {  // But wait, we try to modify the build without loading the execution
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            b.setDescription("Bob");
+            b.save();  // Will trigger an IOException potentially
+        });
+        story.then( r-> {  // Verify that the FlowExecutionOwner can trigger lazy-load correctly
+            WorkflowJob p = r.jenkins.getItemByFullName("p", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            Assert.assertEquals("Bob", b.getDescription());
+            Assert.assertEquals("4", b.getExecution().getCurrentHeads().get(0).getId());
```

Comment:
Asserting a particular flow node ID is very fragile. Please change the assertion to verify something intrinsically meaningful. Say, that it is a `FlowEndNode`, or that its parent is an `echo` step, or something.

(Did I not suggest in some old PR that there be some kind of test utility to assert the nodes in a build? Something in `workflow-api-*-tests.jar` would be ideal.)

Reply:
@jglick So... it's only fragile if we change how iotas are done, which at this point would break a lot of things (there was a time when we could have done that without much impact, but it was years ago).    But some of those sorts of structural tests can be done by combining filter Predicates with the FlowScanners pretty trivially. 

For a case like this though, we really *do* need to assert that the ID is correct because that ensures no sort of placeholder FlowNodes have been inserted.